### PR TITLE
Support escape path for MacOs, Linux

### PR DESCRIPTION
**Pull request by khanhntd**
# Description of the issue
Before the changes, CloudWatchAgent does not able to use escape path for both Mac and Linux while this does not apply to Windows. 

# Description of changes
There are only 1 major change:
* Add single quote and use echo to execute the command  and also show the command output the same time

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
**Before the changes:**
* **On Mac:** By running the following command `sudo ./amazon-cloudwatch-agent-ctl -a fetch-config -c file:"./Test Dir/config.json"`, the expect result is
```
/opt/aws/amazon-cloudwatch-agent/bin/config-downloader --output-dir /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d --download-source file:./Test Dir/config.json --mode auto --config /opt/aws/amazon-cloudwatch-agent/etc/common-config.toml --multi-config default
2022/06/24 13:41:10 E! ec2metadata is not available
I! detect region from ecs
Unable to determine aws-region.
Please check if you can access the metadata service. For exampe, on linux, run 'wget -q -O - http://169.254.169.254/latest/meta-data/instance-id && echo' 
Fail to fetch the config!
```

* **On Linux:** By running the following command `sudo ./amazon-cloudwatch-agent-ctl -a fetch-config -c file:"/home/khanhntd/Code/CloudWatch/Test Dir/config.json"`, the expect result is
```
****** processing amazon-cloudwatch-agent ******
/opt/aws/amazon-cloudwatch-agent/bin/config-downloader --output-dir /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d --download-source file:/home/khanhntd/Code/CloudWatch/Test Dir/config.json --mode ec2 --config /opt/aws/amazon-cloudwatch-agent/etc/common-config.toml --multi-config default
I! Trying to detect region from ec2
2022/06/24 17:21:38 D! [EC2] Found active network interface
2022/06/24 17:21:38 /home/khanhntd/Code/CloudWatch/Test
2022/06/24 17:21:38 test path%!(EXTRA string=file_Test)
2022/06/24 17:21:38 E! Fail to fetch/remove json config: open /home/khanhntd/Code/CloudWatch/Test: no such file or directory
E! Fail to fetch/remove json config: open /home/khanhntd/Code/CloudWatch/Test: no such file or directory
Fail to fetch the config!
```

**After the changes**
* **On Mac:** By running the following command `sudo ./amazon-cloudwatch-agent-ctl -a fetch-config -c file:"./Test Dir/config.json"`, the expect result is
```
2022/06/24 12:20:32 E! ec2metadata is not available
I! Detected the instance is OnPrem
Got Home directory: /var/root
I! Set home dir Linux: /var/root
I! SDKRegionWithCredsMap region:  us-east-1
Successfully fetched the config and saved in /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d/file_config.json.tmp
/opt/aws/amazon-cloudwatch-agent/bin/config-translator --input /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.json --input-dir /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d --output /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.toml --mode auto --config /opt/aws/amazon-cloudwatch-agent/etc/common-config.toml --multi-config default
/opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent -schematest -config /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.toml
Configuration validation second phase failed
======== Error Log ========
2022-06-24T16:20:32Z E! [telegraf] Error running agent: Agent flush_interval must be positive; found 1m0s
```
* **On Linux:** By running the following command ` sudo ./amazon-cloudwatch-agent-ctl -a fetch-config -c file:"/home/khanhntd/Code/CloudWatch/Test Dir/config.json"`, the expect result is
```
****** processing amazon-cloudwatch-agent ******
2022/06/24 16:43:55 /home/khanhntd/Code/CloudWatch/Test Dir/config.json
2022/06/24 16:43:55 test path%!(EXTRA string=file_config.json)
Got Home directory: /root
I! Set home dir Linux: /root
I! SDKRegionWithCredsMap region:  us-west-2
Successfully fetched the config and saved in /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d/file_config.json.tmp
Start configuration validation...
2022/06/24 16:43:55 Reading json config file path: /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d/file_config.json.tmp ...
2022/06/24 16:43:55 I! Valid Json input schema.
2022/06/24 16:43:55 D! [EC2] Found active network interface
I! Detecting run_as_user...
Got Home directory: /root
I! Set home dir Linux: /root
I! SDKRegionWithCredsMap region:  us-west-2
No csm configuration found.
Configuration validation first phase succeeded
/opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent -schematest -config /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.toml
Configuration validation second phase succeeded
Configuration validation succeeded
```

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make linter`

### CODE REVIEW

**Comment by khanhntd** [comment]
> Can you please show the output for what the behavior is before your change?

Yes I have attached the output before the changes for both Linux and MacOS

File: translator/jsonconfig/mergeJsonConfig.go [hunk 1]
```diff
@@ -22,15 +21,14 @@ func MergeJsonConfigMaps(jsonConfigMapMap map[string]map[string]interface{}, def
 		if os.Getenv(config.USE_DEFAULT_CONFIG) == config.USE_DEFAULT_CONFIG_TRUE {
 			// When USE_DEFAULT_CONFIG is true, ECS and EKS will be supposed to use different default config. EKS default config logic will be added when necessary
 			if ecsutil.GetECSUtilSingleton().IsECS() {
-				fmt.Println("No json config files found, use the default ecs config")
+				log.Println("No json config files found, use the default ecs config")
```
Comment (SaxyPandaBear) [inline translator/jsonconfig/mergeJsonConfig.go:24]: Are we _sure_ about doing this change? The last time that we switched from fmt to log, it caused a regression.

**Review by SaxyPandaBear — COMMENTED** [review]
One thing that isn't shown is test output for a case where there is not spaces in the file path. This feels like an obvious case, but please show evidence that there isn't a regression for that.

File: translator/jsonconfig/mergeJsonConfig.go (hunk 1 above)
Comment (khanhntd) [reply translator/jsonconfig/mergeJsonConfig.go:24]: Yes. There is one reason why I introduced this change:
* Even though I have inputed the wrong json format, CWAgent still say `2022/06/24 16:42:59 I! Valid Json input schema.` (under the hood, if there is a wrong json format, CWAgent will use the default one to override it)

This changes only applied to the `config-translator` binaries and after watching the [shell](https://github.com/aws/amazon-cloudwatch-agent/blob/master/packaging/windows/amazon-cloudwatch-agent-ctl.ps1#L377), I believe there won't be any regression error.

**Comment by khanhntd** [comment]
Here are the test without space:
* On Mac:
```
khanhntd@88665a0eaf54 darwin % sudo ./amazon-cloudwatch-agent-ctl -a fetch-config -c file:"./config.json"
2022/06/24 18:06:56 E! ec2metadata is not available
I! Detected the instance is OnPrem
Got Home directory: /var/root
I! Set home dir Linux: /var/root
I! SDKRegionWithCredsMap region:  us-east-1
Successfully fetched the config and saved in /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d/file_config.json.tmp
2022/06/24 18:06:56 D! [EC2] Found active network interface
2022/06/24 18:07:02 E! ec2metadata is not available
2022/06/24 18:07:02 Reading json config file path: /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d/file_config.json.tmp ...
I! Detected the instance is OnPrem
Valid Json input schema.
Got Home directory: /var/root
I! Set home dir Linux: /var/root
I! SDKRegionWithCredsMap region:  us-east-1
No csm configuration found.
Under path : /logs/ | Info : Got hostname 88665a0eaf54.ant.amazon.com as log_stream_name
Configuration validation first phase succeeded
/opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent -schematest -config /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.toml
Configuration validation second phase succeeded
Configuration validation succeeded

```
* On Linux:
```
dev-dsk-khanhntd-1e-76a8167a % sudo ./amazon-cloudwatch-agent-ctl -a fetch-config -c file:"/home/khanhntd/Code/CloudWatch/config.json"
****** processing amazon-cloudwatch-agent ******
2022/06/24 22:01:26 /home/khanhntd/Code/CloudWatch/config.json
2022/06/24 22:01:26 test path%!(EXTRA string=file_config.json)
Got Home directory: /root
I! Set home dir Linux: /root
I! SDKRegionWithCredsMap region:  us-west-2
Successfully fetched the config and saved in /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d/file_config.json.tmp
Start configuration validation...
2022/06/24 22:01:26 test again
2022/06/24 22:01:26 test
2022/06/24 22:01:26 test again
2022/06/24 22:01:26 test more
2022/06/24 22:01:26 Reading json config file path: /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d/file_config.json.tmp ...
2022/06/24 22:01:26 unable to scan config dir /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d with error: unable to parse json, error: invalid character '}' looking for beginning of object key string
2022/06/24 22:01:26 map[]
2022/06/24 22:01:26 test again %v %v 0 false
2022/06/24 22:01:26 No json
2022/06/24 22:01:26 I! Valid Json input schema.
2022/06/24 22:01:26 D! [EC2] Found active network interface
No json config files found, use the default one
I! Detecting run_as_user...
Got Home directory: /root
I! Set home dir Linux: /root
I! SDKRegionWithCredsMap region:  us-west-2
No csm configuration found.
No log configuration found.
Configuration validation first phase succeeded
/opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent -schematest -config /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.toml
Configuration validation second phase succeeded
Configuration validation succeeded
```

**Comment by SaxyPandaBear** [comment]
```
 file:"/home/khanhntd/Code/CloudWatch/config.json"
```
Does this mean that we have to supply `"` for this from now on? That is a breaking change.

**Comment by khanhntd** [comment]
> ```
>  file:"/home/khanhntd/Code/CloudWatch/config.json"
> ```
> 
> Does this mean that we have to supply `"` for this from now on? That is a breaking change.

No you don't have too. " is only needed for spaces. While you can use without " if there does not have any space.

**Review by adam-mateen — APPROVED** [review]
approved.
Thanks for sharing output from manual testing.

**Review by SaxyPandaBear — APPROVED** [review]

**Pull request merged**